src/main/java/ai/promoted/delivery/client/ApiDelivery.java

```diff
@@ -75,9 +102,19 @@ public ApiDelivery(String endpoint, String apiKey, long timeoutMillis, boolean w
 
     this.httpClient = HttpClient.newBuilder().version(HttpClient.Version.HTTP_2).build();
     
+    // The gRPC path doesn't support health checks or warm-up yet, so we re-use the HTTP path.
     if (warmup) {
       runWarmup();
     }
+
+    if (useGrpc) {
```

Comment:
If clients object to grpc, they probably don't want the dependency in their binary.

Long-term, I'd either do:
1. moving everyone over to grpc.
2. having different builds for grpc vs json.

If you want it temporarily, that's fine.

Reply:
Yeah, a more proper thing here (for the ecosystems I'm familiar with) would be conditional builds but it either doesn't seem like a thing in Java or it gets messy. I can see users needing additional functionality added to the SDK but not being able to move to gRPC so I don't want to require a move at this point.